build_resolvers/lib/src/resolver.dart

```diff
@@ -304,23 +305,31 @@ List<int> _buildSdkSummary(String dartSdkPath) {
 /// [sdk].
 void _addFlutterLibraries(AbstractDartSdk sdk,
     ResourceProvider resourceProvider, String dartSdkPath) {
-  var embedderSdkPath =
-      p.join(p.dirname(dartSdkPath), 'pkg', 'sky_engine', 'lib');
   var embedderYamlFile =
-      resourceProvider.getFile(p.join(embedderSdkPath, '_embedder.yaml'));
+      resourceProvider.getFile(p.join(_dartUiPath, '_embedder.yaml'));
   if (!embedderYamlFile.exists) {
     throw StateError('Unable to find flutter libraries, please run '
         '`flutter precache` and try again.');
   }
 
   var embedderYaml = loadYaml(embedderYamlFile.readAsStringSync()) as YamlMap;
   var flutterSdk = EmbedderSdk(resourceProvider,
-      {resourceProvider.getFolder(embedderSdkPath): embedderYaml});
+      {resourceProvider.getFolder(_dartUiPath): embedderYaml});
 
   for (var library in flutterSdk.sdkLibraries) {
     if (sdk.libraryMap.getLibrary(library.shortName) != null) continue;
     sdk.libraryMap.setLibrary(library.shortName, library as SdkLibraryImpl);
   }
 }
 
-final _isFlutter = Platform.version.contains('flutter');
+/// Path to the running dart's SDK root.
+final _runningDartSdkPath = p.dirname(p.dirname(Platform.resolvedExecutable));
+
+/// Path where the dart:ui package will be found, if executing via the dart
+/// binary provided by the Flutter SDK.
+final _dartUiPath = p.canonicalize(
```

Comment:
nit: use `p.normalize` instead of `p.canonicalize` - we don't need the canonical (absolute) path just a normalized one :).

Reply:
Done.